packages/core/src/renderer/components/cluster-settings/cluster-settings-menu-clear-item.injectable.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import type { ClusterPreferences } from "../../../common/cluster-types";
+import type { ChangedClusterPreference } from "./cluster-settings-menu-injection-token";
+import { clusterIconSettingsMenuInjectionToken } from "./cluster-settings-menu-injection-token";
+
+const clusterIconSettingsMenuClearItem = getInjectable({
+  id: "cluster-icon-settings-menu-clear-item",
+
+  instantiate: () => ({
+    id: "clear-icon-menu-item",
+    title: "Clear",
+    disabled: (preferences: ClusterPreferences) => !preferences.icon,
```

Comment:
Why only `ClusterPreferences` are passed as arguments? What if to determine state we need some more from the `Cluster`?
Also if you're feel free to pass `preferences` to `disabled()` why not pass same way to `onClick()` and did the update of preferences directly inside `onClick(evt, preferences)`?

Reply:
> Why only ClusterPreferences are passed as arguments? What if to determine state we need some more from the Cluster?

Still under consideration what should be passes to injectables. This may vary from specific needs of this part of an application.

> and did the update of preferences directly inside onClick(evt, preferences)?

Probably good idea, refactoring...

Reply:
Maybe passing `clusterId` would be good enough? That way we have the specifics of where this is stored not be part of the contract here?

Reply:
Ended up passing preferences to `onClick` event and changing them right there inside injectable.